scanpy/testing/_pytest/marks.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import annotations
+from importlib.util import find_spec
+
+import pytest
+
+
+def make_skip_mark(mod: str, dist: str | None = None):
+    reason = f"needs module `{mod}`"
+    if dist:
+        reason = f"{reason} (`pip install {dist}`)"
+    return pytest.mark.skipif(not find_spec(mod), reason=reason)
+
+
+needs_leidenalg = make_skip_mark("leidenalg")
+needs_louvain = make_skip_mark("louvain")
+needs_skmisc = make_skip_mark('skmisc', 'scikit-misc')
+needs_scanorama = make_skip_mark("scanorama")
+needs_scrublet = make_skip_mark("scrublet")
+needs_fa2 = make_skip_mark("fa2")
+needs_igraph = make_skip_mark("igraph", "python-igraph")
```

Comment:
They’re defined like this to be reused. They need to be reused because some module names don’t match the distribution/PyPI names.

What kind of documentation do you mean? A page in the “Contributing / Tests” section explaining what marks and fixtures we have? If you mean a docstring or comment, then I’d say that I think the skip marks are pretty self explanatory.

Reply:
> They’re defined like this to be reused.

Sure, but I would prefer something like: `@skip_if_uninstalled("scanorama")`. It's easier to see how to extend that to new cases.

------

A docstring saying why we're not using `importorskip` would be good.

There is also already a doc page for more general notes: https://scanpy.readthedocs.io/en/stable/dev/testing.html

Reply:
And `skip_if_uninstalled` uses a hardcoded list of all modules and refuses to work with others so we can make sure the error message names the correct thing? Sure that’d also work. I guess I don’t see the advantage of that compared with a bunch of marks, though.

> A docstring saying why we're not using `importorskip` would be good.

makes sense!

Reply:
Why do you think a `skip_if_uninstalled` if better than the individual marks?

Reply:
OK, so it’s impossible to use [`importlib.metadata.packages_distributions`](https://docs.python.org/3/library/importlib.metadata.html#mapping-import-to-distribution-packages) since we’re dealing with things that *aren’t* installed. So unless we want our tests to talk to the PyPI API, we should hardcode those.